components/snowflake/sources/common-table-scan.js

```diff
@@ -0,0 +1,142 @@
+const { isNil } = require("lodash");
+const snowflake = require("../snowflake.app");
+
+module.exports = {
+  props: {
+    snowflake,
+    db: "$.service.db",
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15, // 15 minutes
+      },
+    },
+    tableName: {
+      type: "string",
+      label: "Table Name",
+      description: "The name of the table to watch for new rows",
+      async options(context) {
+        const { page } = context;
+        if (page !== 0) {
+          return [];
+        }
+
+        const options = await this.snowflake.listTables();
+        return options.map(i => i.name);
+      },
+    },
+    uniqueKey: {
+      type: "string",
+      label: "Unique Key",
+      description: "The name of a column in the table to use for deduplication. Defaults to `ID`",
+      default: "ID",
+      async options(context) {
+        const { page } = context;
+        if (page !== 0) {
+          return [];
+        }
+
+        const options = await this.snowflake.listFieldsForTable(this.tableName);
+        return options.map(i => i.name);
+      },
+    },
+    eventSize: {
+      type: "integer",
+      label: "Event Size",
+      description: "The number of rows to include in a single event (by default, emits 1 event per row)",
+      default: 1,
+      min: 1,
+    },
+  },
+  hooks: {
+    async activate() {
+      let lastResultId = this.db.get("lastResultId");
+      if (isNil(lastResultId)) {
```

Comment:
this is just `lastResultId == null` right?

Reply:
we should try to minimize deps especially in cases like these

Reply:
`null` or `undefined`, but nevertheless I think this was kept by error.
I'll change it to a comparison against `undefined` and remove the dependency.